### ISSUE TITLE
Question on KRAKEN module

### ISSUE THREAD

**Issue by zl945** [bxlab/metaWRAP#133]
Through Assembly module, I got 82899 contigs. But when I annotated contigs with the KRAKEN module, it turned out that only 6,161 contigs were annotated. why?

metawrap kraken -o KRAKEN -t 18 final_assembly.fasta 

![1551076791 1](https://user-images.githubusercontent.com/47781232/53320081-74e57c00-390f-11e9-8880-c7305ecf92a9.jpg)

**Comment by ursky** [comment]
While low annotation fraction is pretty common, this is in fact quite low for contigs. Here are some reasons that this could be the case:
1. The vast majority of the contigs were too short (<1kb) to be annotated. Can you look at your assebly file to check?
2. The assembly quality is very poor due to bad sequencing, resulting in erroneous sequence. How much sequencing went into this? What is the N50 of the assembly? over 80k contigs is a VERY large number, even for a metagenomic assembly.
3. You are sequencing something that is so novel that it shares very little sequence with the database (unlikely).

**Comment by zl945** [comment]
assembly_report：
![1551168184](https://user-images.githubusercontent.com/47781232/53396724-2522b580-39e0-11e9-9406-3c2a22a7c2d2.jpg)

**Comment by ursky** [comment]
Thanks, this report is very telling. You have a huge number of short fragments (1-5kb) in the assembly - more than I usually see from typical libraries. I do not know the details and context of the community, library, and sequencing, but these short contigs are likely erroneous sequences, which is why they cannot be assembled or classified effectively. You had ~6000 annotated contigs, which roughly corresponds to the number of contigs >5kb. If it were me I would look at the Nx plot (in the same report you sent) of the assembly, chose a higher length threshold (2-5kb, depending on what the curve looks like), and proceed with only these longer contigs.

**Comment by zl945** [comment]
I agree with you. It may be caused by erroneous sequences. Can you give me some suggestions, for one thing, how can i reduce the numers of erroneous sequences of contigs and classify them effectively. For another, Can you tell me how this moudle works? or What problems should I pay attention to when I use this moudle? Thanks in advance.

**Comment by ursky** [comment]
Like I said, the best you can probably do is throw away short contigs that are likely not meaningful. Look at the Nx plot to decide on a minimum (probably 3-5kb). Perhaps trying with a different assembler would help as well (if you used metaspades try megahit, or vice versa).

You can find the descriptions of all modules at https://github.com/bxlab/metaWRAP/blob/master/Module_descriptions.md
